core/src/main/scala/org/apache/spark/memory/UnifiedMemoryManager.scala

```diff
@@ -187,7 +187,6 @@ object UnifiedMemoryManager {
   // This serves a function similar to `spark.memory.fraction`, but guarantees that we reserve
   // sufficient memory for the system even for small heaps. E.g. if we have a 1GB JVM, then
   // the memory used for execution and storage will be (1024 - 300) * 0.75 = 543MB by default.
-  private val RESERVED_SYSTEM_MEMORY_BYTES = 300 * 1024 * 1024
```

Comment:
Let me back up a second -- this minimum is really specific to the new memory manager. Actually, should it apply to the old memory manager? That is, what error were you referring to initially? I had assumed that you mean that, even in legacy mode, this limit is later applied but in some confusing way, but I'm not as sure that was the problem. If it really is a required minimum even when using legacy mode, we should fail faster; if it's not actually being required now in legacy mode, then we should fix whatever thinks it is.